### PR TITLE
Dev icmp

### PR DESCRIPTION
**Pull request by shramroy**

### CODE REVIEW

**Review by ajbalogh — CHANGES_REQUESTED** [review]
Keep the top level Flow.Icmp schema object.
Expand the choice with specific schema objects for non deprecated/reserved/unassigned types.
Each specific schema object has the correct named fields.
Each schema object code field has the correct x-constants.
Initially only create schema objects for headers that have been asked for.

Naming convention:
Flow.IcmpMsg0
- codes and fields specific to this msg
Flow.IcmpMsg3
- codes and fields specific to this msg
Flow.IcmpMsg5
- codes and fields specific to this msg

If you need more info please setup a meeting with me (slack/teams) and we can go over it together.

**Comment by shramroy** [comment]
I have modified the model changes for icmp according the requested changes.

File: flow/packet-headers/icmp.yaml [hunk 1]
```diff
@@ -39,9 +33,14 @@ components:
             description: >-
               code of ICMP with next fields
             format: integer
-            length: 8
+            length: 1
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:36]: The length of code is 8 bits. The description should include a comment that a code of 0 is used by both echo request and echo reply.

File: flow/packet-headers/icmp.yaml [hunk 2]
```diff
@@ -39,9 +33,14 @@ components:
...
             default: 0
             features: [count, metric_group]
+
+    Flow.ICMP.NextFields:
+      description: Next Fields for ICMP that contains ID, Seq Number and Next Fields
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:41]: Identifier and sequence number are 16bits each which makes up the rest of header.
The remainder of the packet is variable data up to a max of 576 bytes which can be filled in using the custom packet header which means next fields should not be present.

File: flow/packet-headers/icmpv6.yaml [hunk 3]
```diff
@@ -46,7 +46,7 @@ components:
             description: >-
               code of ICMPv6 Destination Unreachable Message
             format: integer
-            length: 8
+            length: 3
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:49]: Length is 8 bits. All x-constants should be lowercase with underscores.

File: flow/packet-headers/icmpv6.yaml [hunk 4]
```diff
@@ -81,7 +81,7 @@ components:
             description: >-
               code of ICMPv6 Packet Too Big Message
             format: integer
-            length: 8
+            length: 1
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:84]: Length is 8 bits.

File: flow/packet-headers/icmpv6.yaml [hunk 5]
```diff
@@ -158,7 +158,7 @@ components:
             description: >-
               code of ICMPv6 Parameter Problem
             format: integer
-            length: 8
+            length: 2
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:161]: Length is 8 bits.

**Review by ajbalogh — CHANGES_REQUESTED** [review]

**Comment by shramroy** [comment]
@ajbalogh I have pushed all the changes. Kindly review it once.

File: flow/packet-headers/icmp.yaml [hunk 6]
```diff
@@ -7,70 +7,73 @@ components:
...
+        echo:
+          $ref: '#/components/schemas/Flow.Icmp.Echo'
 
-    Flow.ICMPEcho:
+    Flow.Icmp.Echo:
       x-include: 
-      - '.#/components/schemas/Flow.ICMP.NextFields'
-      description: Packet Header for ICMP Echo
+      - '.#/components/schemas/Flow.Icmp.Common'
+      - '.#/components/schemas/Flow.Icmp.NextFields'
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:17]: beginning . is not needed and should be removed.

File: flow/packet-headers/icmpv6.yaml [hunk 7]
```diff
@@ -7,311 +7,64 @@ components:
...
-              sequence number of ICMPv6 Echo Reply
+              ICMPv6 echo sequence number
             format: integer
             length: 16
             default: 0
             features: [count, metric_group]
 
-    Flow.ICMPv6MulticastListnerQuery:
-      description: Packet Header for ICMPv6 Multicast Listner Query
+    Flow.Icmp.Common:
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:56]: This is the same name as in icmp.yaml - it should be changed to Flow.Icmpv6.Common

File: flow/packet-headers/icmpv6.yaml [hunk 8]
```diff
@@ -7,311 +7,64 @@ components:
...
-          $ref: '#/components/schemas/Flow.ICMPv6MulticastListnerQuery'
+          enum: [echo]
+        echo:
+          $ref: '#/components/schemas/Flow.Icmpv6.Echo'
         
-    Flow.ICMPv6DestUnreach:
-      description: Packet Header for ICMPv6 Destination Unreachable Message
+    Flow.Icmpv6.Echo:
+      x-include: 
+      - '.#/components/schemas/Flow.Icmp.Common'
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:16]: this references the same name as in icmp.yaml - it should reference a unique name in this file
beginning . should be removed

**Review by ajbalogh — CHANGES_REQUESTED** [review]
lgtm - just a few changes needed

File: flow/packet-headers/icmp.yaml (hunk 6 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml]: Removed the dots

File: flow/packet-headers/icmpv6.yaml (hunk 7 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml]: Changed the name

File: flow/packet-headers/icmpv6.yaml (hunk 8 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml]: Changed the name

**Review by ajbalogh — APPROVED** [review]

**Pull request merged**